types/react-beautiful-dnd/index.d.ts

```diff
@@ -34,73 +174,124 @@ export interface DropResult {
+    droppableId: DroppableId;
+    index: number;
 }
 
-export class DragDropContext extends React.Component<DragDropContextProps> {}
-
 /**
- *  Droppable
+ *  Draggable
  */
+export interface DraggingStyle {
+    // `position: fixed` is used to ensure that the element is always positioned
+    // in the correct position and ignores the surrounding position:relative parents
+    position: 'fixed';
 
-export interface DroppableProvided {
-    innerRef(element: HTMLElement | null): any;
-    placeholder?: React.ReactElement<any>;
+    // When we do `position: fixed` the element looses its normal dimensions,
+    // especially if using flexbox. We set the width and height manually to
+    // ensure the element has the same dimensions as before it started dragging
+    width: number;
+    height: number;
+
+    // The width and height values take into account whether the original element
+    // used `box-sizing: content-box` or `box-sizing: border-box`
+    // Because we are setting the width and height directly we want to ensure that
+    // these are the actual values applied
+    boxSizing: 'border-box';
+
+    // We initially position the element in the same *visual spot* as when it started.
+    // This means that these values *exclude* the original margins so that element remains
+    // in the same visual position - even though now it has no margins
+    top: number;
+    left: number;
+
+    // We clear any top or left margins on the element to ensure it does not push
+    // the element positioned with the top/left position (which is margin aware).
+    // We also clear the margin right / bottom. This has no positioning impact,
+    // but it is cleanest to just remove all the margins rather than only the top and left.
+    margin: 0;
+
+    // We need to opt out of the shared global style that is being applied to
+    // all draggables. The movement of moving draggables is either not animated
+    // or handled by react-motion.
+    transition: 'none';
+
+    // Move the element in response to a user dragging
+    transform?: Maybe<string>;
+
+    // When dragging or dropping we control the z-index to ensure that
+    // the layering is correct
+    zIndex: ZIndex;
 }
 
-export interface DroppableStateSnapshot {
-    isDraggingOver: boolean;
+export interface NotDraggingStyle {
+    transform?: Maybe<string>;
+    // null: use the global animation style
+    // none: skip animation (used in certain displacement situations)
+    transition: null | 'none';
 }
 
-export interface DroppableProps {
-    droppableId: DroppableId;
-    type?: TypeId;
-    isDropDisabled?: boolean;
-    direction?: 'vertical' | 'horizontal';
-    children(provided: DroppableProvided, snapshot: DroppableStateSnapshot): React.ReactElement<any>;
+export type DraggableStyle = DraggingStyle | NotDraggingStyle;
+
+// Props that can be spread onto the element directly
+export interface DraggableProps {
+    // inline style
+    style?: Maybe<DraggableStyle>;
```

Comment:
This type including `null` is quite unfortunate, as `null` is not an allowed value for React element's `style` prop. This means that you can't simply spread a `DraggableProps` on an element.

![image](https://user-images.githubusercontent.com/105820/35079952-b83c15b6-fc5e-11e7-8111-08397e4a2edd.png)